kroxylicious-api/src/main/java/io/kroxylicious/proxy/metadata/DescribeTopicLabelsResponse.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.metadata;
+
+import java.util.Map;
+
+/**
+ * A response to a {@link DescribeTopicLabelsRequest} detailing the labels of each of the topics.
+ * @param topicLabels
+ */
+public record DescribeTopicLabelsResponse(Map<String, Map<String, String>> topicLabels) implements ResourceMetadataResponse<DescribeTopicLabelsRequest> {
```

Comment:
I think we should probably define a wrapper around the map. Or provide query methods on this record. Handing round nested maps feels like it could be trouble for evolving the API.

Reply:
I know what you're getting at. But:

* what methods should we actually expose? Do we just hide the existence of the outer map? It's difficult to also hide the inner maps while also providing an ergonomic API, because, well, labels are conceptually exactly a `Map`. But if it's OK to expose the map-ness of the inner maps, why isn't it OK to expose the map-ness of the outer map? 
* But if we expose inner maps should they be modifiable? If not, where do we make them unmodifiable? 
* If we worried about evolving the API might it be better to return an interface, so filter author's test code can't couple to the constructor signature? But that forces them into using mocks, which isn't much fun for them.

I don't immediately see a consistent and principled way through these questions, but happy to listen to ideas.